java-frontend/src/test/java/org/sonar/java/regex/ast/BackReferenceTreeTest.java

```diff
@@ -19,70 +19,90 @@
  */
 package org.sonar.java.regex.ast;
 
+import java.util.ArrayList;
+import java.util.List;
 import org.junit.jupiter.api.Test;
 
+import static org.apache.commons.lang.StringEscapeUtils.escapeJava;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.sonar.java.regex.RegexParserTestUtils.assertFailParsing;
-import static org.sonar.java.regex.RegexParserTestUtils.assertKind;
 import static org.sonar.java.regex.RegexParserTestUtils.assertSuccessfulParse;
 
 class BackReferenceTreeTest {
 
   @Test
   void backReferences() {
-    assertBackReference("\\\\k<group1>", "group1");
-    assertBackReference("\\\\k<ALPHA>", "ALPHA");
-    assertBackReference("\\\\k<0invalid>", "0invalid");
+    assertBackReference(escapeJava("\\k<group1>"), "group1");
+    assertBackReference(escapeJava("\\k<ALPHA>"), "ALPHA");
+    assertBackReference(escapeJava("\\k<0invalid>"), "0invalid");
 
-    assertBackReference("\\\\1", 1);
```

Comment:
I'd like to keep this one as it's the most simple test possible and unlike the next one, `\\1` should indeed still be parsed as a single backreference.